git_connector.py

```diff
@@ -485,10 +504,24 @@ def _clone_repo(self, param):
 
         action_result = self.add_action_result(ActionResult(dict(param)))
 
-        repo_dir = self.app_state_dir / self.repo_name
+        if not self.get_config().get("repo_uri") and not param.get("repo_url", None):
+            msg = "You must either provide a URL to clone or configure the app with information"
+            return action_result.set_status(phantom.APP_ERROR, msg)
+
+        if not param.get("repo_url", None):
+            repo_dir = self.app_state_dir / self.repo_name
+        else:
+            self.modified_repo_uri = param.get("repo_url")
```

Comment:
In this implementation, when cloning the repo in try except block below, it is using the branch name  configured in asset configuration i.e `self.branch_name.`

In case `repo uri` mentioned in test connectivity has a branch name `master` but while running action `clone_repo` a public repo is given in action parameter then it will look for the `master` branch in public repo too which is not correct. That branch may or may not be present in the public repo. And to get this action passed in that case, the user will have to change branch name in asset parameter.

We will suggest that if you are providing support for public repo then you can add parameter for public repo branch at action level too.